File: learnAPI/7.r
library(jsonlite)
library(dplyr)
library(httr)
library(tibble)
library(tidyr)

url <- "https://official-joke-api.appspot.com/random_joke"

response <- GET(url)

if (http_status(response)$category == "Success") {
    json_content <- content(response, "text")
    data <- fromJSON(json_content)

    df <- t(as.data.frame(data))
    colnames(df) <- 'Joke'
    print(head(df))
}

File: learnAPI/7.py
import requests
import json
import pandas as pd

url = "https://official-joke-api.appspot.com/random_joke"

response = requests.get(url)

if response.status_code == 200:
    data = json.loads(response.text)
    # data = response.json()

    
    df = pd.DataFrame({
        'Joke': data
        })
    print(df)
    
    df = pd.DataFrame({
        1: data
        })
    print(df)
    
    df = pd.DataFrame({
        1: data
        }).T
    print(df)
else:
    print("Failed to retrieve data:", response.status_code)